### ISSUE TITLE
Fix ServiceCall raw type reference in createResourceAsync

### ISSUE THREAD

**Issue by anuchandy** [Azure/azure-sdk-for-java#1017]
We have updated serviceCall to a generic type, this means we should not use it as a raw type anywhere but we do in createResourceAsync.

**Comment by martinsawicki** [comment]
btw - I already fixed this in Networking - as part of my LB PR - just adding the type params as you suggested.

**Comment by anuchandy** [comment]
@martinsawicki was thinking more about it and found the fix I suggested is in-correct. If you remember you raised a concern because the fix returns inner, you are right it shouldn't be Inner but the fluent model (or it's base type). 

Let me explain:

We need to first update [this] (https://github.com/Azure/azure-sdk-for-java/blob/master/azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/model/implementation/CreatorTaskGroup.java#L25)

```java
interface ResourceCreator<T> {
        T createResource();
        ServiceCall createResourceAsync(ServiceCallback<T> serviceCallback);
        ...
}
```
to

```java
interface ResourceCreator<T> {
       T createResource(); 
       ServiceCall<T> createResourceAsync(ServiceCallback<T> serviceCallback);
        ...
}
```

i.e. `ServiceCall => ServiceCall<T>`

The graph is a HashMap of `ResourceCreator<Resource>`, we use strongly typed HastMap so values in it needs to be strongly typed hence `ResourceCreator<Resource>`.

Each fluent model (VirtualMachineImpl, StorageAccountImpl) implements this interface, since the fluent models are of `Resource`, type can return `this` from the interface implementation:

```java
Resource StorageAccountImpl::createResource() {
  ...
  return this;
}
```

This means the corresponding async version needs to:

```java
ServiceCall<Resource> StorageAccountImpl::createResourceAsync(final ServiceCallback<Resource> callback) {
  ... 
}
```

So the correct fix is to return a service call that wraps the ServiceCall returned by the SDK createOrUpdate method call. I did a somewhat similar (not exactly) wrapping [here](https://github.com/anuchandy/azure-sdk-for-java/blob/resources-create/azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/arm/collection/implementation/CreatableResourcesImpl.java#L50)

We can also think of moving such wrappings to a util. 

@jianghaolu @martinsawicki thoughts?
